pkg/datamodel/datamodel.go

```diff
@@ -288,15 +288,11 @@ type Output struct {
 	InstillUIOrder int32  `json:"instillUiOrder,omitempty" yaml:"instill-ui-order,omitempty"`
 }
 
-type On struct {
-	Event    map[string]*Event    `json:"event,omitempty" yaml:"event,omitempty"`
-	Schedule map[string]*Schedule `json:"schedule,omitempty" yaml:"schedule,omitempty"`
-}
-
 type Event struct {
-	Type  string         `json:"type,omitempty" yaml:"type,omitempty"`
-	Event string         `json:"event,omitempty" yaml:"event,omitempty"`
-	Setup map[string]any `json:"setup,omitempty" yaml:"setup,omitempty"`
+	Type   string         `json:"type,omitempty" yaml:"type,omitempty"`
+	Event  string         `json:"event,omitempty" yaml:"event,omitempty"`
+	Config map[string]any `json:"config,omitempty" yaml:"config,omitempty"`
```

Comment:
Hi @jvallesm,

I’d like to discuss an idea with you.

I’m currently refactoring the run-on-event recipe, which requires both an event-specific configuration and a connection setup. Here’s an example:
```
on:
  <event-id>:
    type: slack
    event: EVENT_NEW_MESSAGE
    config:
      channel: xxx
    setup: ${connection.my-slack}
```
Having both config and setup feels a bit confusing. Would it be clearer if we renamed setup to connection?

```
on:
  <event-id>:
    type: slack
    event: EVENT_NEW_MESSAGE
    config:
      channel: xxx
    connection: ${connection.my-slack}
```
If so, we’ll also need to update the `setup` field to `connection` in our component tasks for consistency. What do you think?